wandb/sdk/integration_utils/auto_logging.py

```diff
@@ -33,69 +32,70 @@ def get(self, key: K, default: Optional[V] = None) -> Optional[V]:
         ...  # pragma: no cover
 
 
-class RequestResponseResolver(Protocol):
+class ArgumentResponseResolver(Protocol):
     def __call__(
         self,
-        request: Dict[str, Any],
+        args: Sequence[Any],
+        kwargs: Dict[str, Any],
         response: Response,
         start_time: float,
         time_elapsed: float,
-    ) -> Optional[trace_tree.WBTraceTree]:
+    ) -> Optional[Dict[str, Any]]:
         ...  # pragma: no cover
 
 
-class PatchLLMAPI:
+class PatchAPI:
     def __init__(
         self,
         name: str,
         symbols: Sequence[str],
-        resolver: RequestResponseResolver,
+        resolver: ArgumentResponseResolver,
     ) -> None:
-        """Patches the LLM API to log traces."""
-        # name of the LLM provider, e.g. "Cohere" or "OpenAI"
+        """Patches the API to log wandb Media or metrics."""
+        # name of the LLM provider, e.g. "Cohere" or "OpenAI" or package name like "Transformers"
         self.name = name
-        # api library name, e.g. "cohere" or "openai"
-        self._llm_api = None
+        # api library name, e.g. "cohere" or "openai" or "transformers"
+        self._api = None
         # dictionary of original methods
         self.original_methods: Dict[str, Any] = {}
-        # list of symbols to patch, e.g. ["Client.generate", "Edit.create"]
+        # list of symbols to patch, e.g. ["Client.generate", "Edit.create"] or ["Pipeline.__call__"]
         self.symbols = symbols
-        # resolver callable to convert request/response into a trace tree
+        # resolver callable to convert args/response into a dictionary of wandb media objects or metrics
         self.resolver = resolver
 
     @property
-    def llm_api(self) -> Any:
-        """Returns the LLM API module."""
+    def set_api(self) -> Any:
```

Comment:
maybe we can just call it `api` in this case

Reply:
now that i read the code again maybe module is a better name?
before it was expected to be `llm_api` but if you generalize it doesn't necessarily fit anymore?

Reply:
https://github.com/wandb/wandb/pull/5535#issuecomment-1546076035